ob1k-cache/src/main/java/com/outbrain/ob1k/cache/LoadingCacheDelegate.java

```diff
@@ -98,37 +106,37 @@ public void produce(final Consumer<V> consumer) {
                     return;
                 }
 
-                final ComposableFuture<V> cachedResult = cache.getAsync(key).withTimeout(duration, timeUnit, "LoadingCacheDelegate fetch from cache");
+                final ComposableFuture<V> cachedResult = cache.getAsync(key)
+                  .continueOnError(handleCacheErrors(promise, key))
```

Comment:
ok, looked at it again and it has two bugs in it:

1. in handleCacheErrors() if there is no metric factory you don't call the loader
2. in case you get a timeout from the cache, than call handleCacheErrors() that call the loader and the loader returns null you will call the loader again inside the consumer. reason being that you loose context when using the handleCacheErrors() method i.e. you don't know if you're after the cache or after the loader.

I suggest that the handleCacheErrors() will just transform the TimeoutException to a null result which will trigger the call to the loader inside the consumer.
make sense ?